src/index.js

```diff
@@ -18,40 +19,106 @@ function runPackageScripts({scriptConfig, scripts, args, options = {}}, callback
     scripts = ['default']
   }
   const scriptNames = arrify(scripts)
-  const method = options.parallel ? 'map' : 'mapSeries'
-  async[method](scriptNames, (scriptName, cb) => {
-    const child = runPackageScript({scriptConfig, options, scriptName, args})
-    if (child.on) {
-      child.on('exit', exitCode => cb(null, exitCode))
-    } else {
-      cb(child)
+  if (options.parallel) {
```

Comment:
When running in parallel, you can just use `Promise.all`, when executing serially you can do something like this.

```js
const result = promises.reduce((res, task) => {
   return res.then(() => task());
});

result.then(...);
```